res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1600 @@
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+var DDJSR2 = {};
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// Sets the jogwheels sensivity. 1 is default, 2 is twice as sensitive, 0.5 is half as sensitive.
+DDJSR2.jogwheelSensitivity = 2;
+
+// Sets how much more sensitive the jogwheels get when holding shift.
+// Set to 1 to disable jogwheel sensitivity increase when holding shift (default: 10).
+DDJSR2.jogwheelShiftMultiplier = 10;
+
+// If true, vu meters twinkle if AutoDJ is enabled (default: true).
+DDJSR2.twinkleVumeterAutodjOn = true;
+// If true, selected track will be added to AutoDJ queue-top on pressing shift + rotary selector,
+// else track will be added to AutoDJ queue-bottom (default: false).
+DDJSR2.autoDJAddTop = false;
+// Sets the duration of sleeping between AutoDJ actions if AutoDJ is enabled [ms] (default: 1000).
+DDJSR2.autoDJTickInterval = 1000;
+// Sets the maximum adjustment of BPM allowed for beats to sync if AutoDJ is enabled [BPM] (default: 10).
+DDJSR2.autoDJMaxBpmAdjustment = 10;
+// If true, AutoDJ queue is being shuffled after skipping a track (default: false).
+// When using a fixed set of tracks without manual intervention, some tracks may be unreachable,
+// due to having an unfortunate place in the queue ordering. This solves the issue.
+DDJSR2.autoDJShuffleAfterSkip = false;
+
+// If true, by releasing rotary selector,
+// track in preview player jumps forward to "jumpPreviewPosition"
+// (default: jumpPreviewEnabled = true, jumpPreviewPosition = 0.3).
+DDJSR2.jumpPreviewEnabled = true;
+DDJSR2.jumpPreviewPosition = 0.3;
+
+// If true, PFL / Cue (headphone) is being activated by loading a track into certain deck (default: true).
+DDJSR2.autoPFL = true;
+
+// DDJSR2.shiftPressed = false;
+DDJSR2.rotarySelectorChanged = false;
+DDJSR2.panels = [false, false]; // view state of effect and sampler panel
+// DDJSR2.shiftPanelSelectPressed = false;
+
+DDJSR2.syncRate = [0, 0, 0, 0];
+DDJSR2.needleSearchTouched = [false, false, false, false];
+DDJSR2.chFaderStart = [null, null, null, null];
+DDJSR2.toggledBrake = [false, false, false, false];
+DDJSR2.scratchMode = [true, true, true, true];
+DDJSR2.wheelLedsBlinkStatus = [0, 0, 0, 0];
+DDJSR2.setUpSpeedSliderRange = [0.08, 0.08, 0.08, 0.08];
+DDJSR2.serato=[0xF0, 0x00, 0x20, 0x7F, 0x50, 0x01, 0xF7];
+DDJSR2.seratoPoll=[0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
+DDJSR2.selectedSamplerBank = 0;
+DDJSR2.fxKnobMSBValue = [0, 0];
+DDJSR2.shiftFxKnobMSBValue = [0, 0];
+
+
+DDJSR2.valueVuMeter = {
+    "[Channel1]_current": 0,
+    "[Channel2]_current": 0,
+    "[Channel3]_current": 0,
+    "[Channel4]_current": 0,
+    "[Channel1]_enabled": 1,
+    "[Channel2]_enabled": 1,
+    "[Channel3]_enabled": 1,
+    "[Channel4]_enabled": 1
+};
```

Comment:
also moving these two into a component or two:
DDJSR2.fx1assignLeds = [0x4C, 0x50, 0x70, 0x54];
DDJSR2.fx2assignLeds = [0x4D, 0x51, 0x71, 0x55];

Reply:
Yeah move that directly into the component. The pattern is clear (fx2 is just fx1 + 1).

Reply:
Indeed. Vumeter is better to be directly connected (connect and disconnect appropiately to not need the enabled setting) and send the value directly (except if you are using the value as "previous value" so that t you don't need to send the same value multiple times and so reduce the communication